frame/balances/src/lib.rs

```diff
@@ -213,7 +213,7 @@ pub mod pallet {
 
 	pub type CreditOf<T, I> = Credit<<T as frame_system::Config>::AccountId, Pallet<T, I>>;
 
-	#[pallet::config]
+	#[pallet::default_config]
```

Comment:
For the sake of simplicity, I would revert any changes to the balances pallet. Let's only alter things in a new example pallet.

Reply:
Also, from this, I am sensing that the syntax is to replace `#[pallet::config]` with `#[pallet::default_config]`. 

My mental model is: 

* `#[pallet::config]` declares the trait as the config trait of the pallet.
* `#[pallet::default_config]` (or `generate_default`, `with_default`) generates the extra trait as well. 

So mixing the two is a tad bit confusing IMO. I would suggest making it clear that the default confi is an extension of `#[pallet::config]`, either by being an extra attribute, or something like: 

`#[pallet::config(with_default)]`.

Actually, I like ^^ the best so far :)

Reply:
yes, I like this